internal/k8s/controllers/node_controller_test.go

```diff
@@ -121,3 +123,109 @@ func TestNodeController(t *testing.T) {
 		}
 	}
 }
+
+func TestNodeReconcilerPredicate(t *testing.T) {
+	t.Parallel()
+
+	p := NodeReconcilerPredicate()
+
+	t.Run("allow delete event pass", func(t *testing.T) {
+		t.Parallel()
+		if got, expected := p.Delete(event.DeleteEvent{}), true; got != expected {
+			t.Fatalf("p.Create(event=%+v) returned %v; expected %v", "any", got, expected)
+		}
+	})
+
+	t.Run("allow create event pass", func(t *testing.T) {
+		t.Parallel()
+		if got, expected := p.Create(event.CreateEvent{}), true; got != expected {
+			t.Fatalf("p.Create(event=%+v) returned %v; expected %v", "any", got, expected)
+		}
+	})
+
+	tests := map[string]struct {
+		event    event.UpdateEvent
+		expected bool
+	}{
+		"default": {
+			event: event.UpdateEvent{
+				ObjectOld: &corev1.Node{},
+				ObjectNew: &corev1.Node{},
+			},
+			expected: false,
+		},
+		"wrong event object old": {
+			event: event.UpdateEvent{
+				ObjectOld: &corev1.Pod{},
+				ObjectNew: &corev1.Node{},
+			},
+			expected: false,
+		},
+		"wrong event object new": {
+			event: event.UpdateEvent{
+				ObjectOld: &corev1.Node{},
+				ObjectNew: &corev1.Pod{},
+			},
+			expected: false,
+		},
+		"label change": {
+			event: event.UpdateEvent{
+				ObjectOld: &corev1.Node{},
+				ObjectNew: &corev1.Node{ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"x": "y"}}}},
+			expected: true,
+		},
+		"spec schedulable change": {
```

Comment:
let's add one where both schedulable and label change

Reply:
done